launch_testing/launch_testing/__init__.py

```diff
@@ -53,19 +54,19 @@ def on_stdout_lines(self, lines):
                 continue
             self.stdout_data.write(line + b'\n')
 
-            # Are we ready to quit?
-            if self.regex_match:
-                self.matched = re.match(self.expected_lines[0], line)
-            else:
-                output_lines = self.stdout_data.getvalue().splitlines()
-                self.matched = output_lines == self.expected_lines
-            if self.matched:
-                # We have enough output to compare; shut down my child
-                for td in self.launch_descriptor.task_descriptors:
-                    if td.name == self.name:
-                        td.terminate()
-                        self.killed = True
-                        return
+        # Are we ready to quit?
```

Comment:
Yep, no way for you to know that. My comment isn't from first principles, but rather the memory of working with @esteve to write the test output-matching support, and finding that the ready-to-quit check needed to be done on a line-by-line basis.

How about putting the non-regex check back inside the loop, conditioned on `self.regex == False`, and leave the regex check outside, conditioned on `self.regex == True`?

We might want to further adopt the pattern of adding to `.regex` files a last line containing `.*`, to allow trailing lines to be ignored (at least in test cases where we don't know exactly how many lines will be output).